File: PCWeb/src/types/banner.ts
/**
 * 轮播图跳转类型
 */
export type BannerLinkType = 'none' | 'product' | 'category' | 'page'

/**
 * 轮播图类型
 */
export interface Banner {
  id: string                           // GUID 主键
  image: string                        // 图片 URL
  linkType: BannerLinkType             // 跳转类型
  linkValue: string                    // 跳转目标（商品ID/分类ID/页面路径）
  sort: number                         // 排序（升序）
  status: 0 | 1                        // 状态：0禁用 1启用
  createTime: string                   // 创建时间
  updateTime: string                   // 更新时间
}

/**
 * 创建轮播图参数
 */
export interface CreateBannerParams {
  image: string
  linkType: BannerLinkType
  linkValue: string
  sort?: number
  status?: 0 | 1
}

/**
 * 更新轮播图参数
 */
export interface UpdateBannerParams extends Partial<CreateBannerParams> {
  id: string
}

/**
 * 轮播图查询参数
 */
export interface BannerQueryParams {
  pageIndex?: number
  pageSize?: number
  status?: 0 | 1
}

File: EasyWechatWeb/EasyWeChatModels/Entitys/Banner/Banner.cs
using SqlSugar;

namespace EasyWeChatModels.Entitys;

/// <summary>
/// 轮播图实体类
/// </summary>
/// <remarks>
/// 用于存储首页轮播图信息，包括图片、跳转链接、排序等
/// </remarks>
[SugarTable("Banner", "轮播图表")]
public class Banner
{
    /// <summary>
    /// 轮播图ID（主键）
    /// </summary>
    [SugarColumn(IsPrimaryKey = true)]
    public Guid Id { get; set; } = Guid.NewGuid();

    /// <summary>
    /// 图片URL
    /// </summary>
    /// <remarks>
    /// 轮播图图片地址，长度限制500字符
    /// </remarks>
    [SugarColumn(Length = 500, ColumnDescription = "图片URL")]
    public string Image { get; set; } = string.Empty;

    /// <summary>
    /// 跳转类型
    /// </summary>
    /// <remarks>
    /// 跳转类型：none-无跳转，product-商品详情，category-分类页面，page-指定页面
    /// </remarks>
    [SugarColumn(ColumnDescription = "跳转类型：none-无，product-商品，category-分类，page-页面")]
    public string LinkType { get; set; } = "none";

    /// <summary>
    /// 跳转目标
    /// </summary>
    /// <remarks>
    /// 跳转目标值，根据LinkType不同而不同：
    /// - none: 无值
    /// - product: 商品ID
    /// - category: 分类ID
    /// - page: 页面路径
    /// </remarks>
    [SugarColumn(Length = 100, IsNullable = true, ColumnDescription = "跳转目标")]
    public string? LinkValue { get; set; }

    /// <summary>
    /// 排序
    /// </summary>
    /// <remarks>
    /// 轮播图显示顺序，数字越小越靠前，默认为0
    /// </remarks>
    [SugarColumn(ColumnDescription = "排序")]
    public int Sort { get; set; } = 0;

    /// <summary>
    /// 状态
    /// </summary>
    /// <remarks>
    /// 轮播图状态：1-启用，0-禁用。默认值为1
    /// </remarks>
    [SugarColumn(ColumnDescription = "状态：1-启用，0-禁用")]
    public int Status { get; set; } = 1;

    /// <summary>
    /// 创建时间
    /// </summary>
    /// <remarks>
    /// 轮播图记录创建时间，默认为当前时间
    /// </remarks>
    [SugarColumn(ColumnDescription = "创建时间")]
    public DateTime CreateTime { get; set; } = DateTime.Now;

    /// <summary>
    /// 更新时间
    /// </summary>
    /// <remarks>
    /// 轮播图信息最后更新时间，可为空
    /// </remarks>
    [SugarColumn(IsNullable = true, ColumnDescription = "更新时间")]
    public DateTime? UpdateTime { get; set; }
}